src/popover/popover.ts

```diff
@@ -0,0 +1,16 @@
+import {Component, Input, ChangeDetectionStrategy} from '@angular/core';
+
+@Component({
+  selector: 'ngb-popover-window',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  host: {'[class]': '"popover " + placement', 'role': 'tooltip'},
```

Comment:
huh? This is just a popover decorator, a component that we would be opening ourselves (user's content go into ng-content) so I'm not sure what you mean...